edu-services/gradebook-service/api/src/java/org/sakaiproject/service/gradebook/shared/GradebookService.java

```diff
@@ -743,4 +743,12 @@ public void finalizeGrades(String gradebookUid)
 	 */
 	@SuppressWarnings("rawtypes")
 	List getViewableSections(String gradebookUid);
+
+	/**
+	 * Return the gradebook but with the GradeMappings. The normal getGradebook(siteId)
+	 * doesn't return the GradeMapping.
+	 * @param id
+	 * @return a gradebook with all the GradeMappings.
+	 */
+	public Object getGradebookWithGradeMappings(Long id);
```

Comment:
OK so I would be making a shared DTO and mapping these hibernate objects to that and back again.